### ISSUE TITLE
data output "azurerm_key_vault_access_policy" isn't capitalised correctly

### ISSUE THREAD

**Issue by dvasdekis** [hashicorp/terraform-provider-azurerm#21360]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.4.4

### AzureRM Provider Version

3.51.0

### Affected Resource(s)/Data Source(s)

azurerm_key_vault_access_policy

### Terraform Configuration Files

```hcl
# Create a Keyvault
resource "azurerm_key_vault" "my-kv" {
  name                        = "my-kv"
  resource_group_name = data.azurerm_resource_group.rg-c.name
  location            = data.azurerm_resource_group.rg-c.location
  soft_delete_retention_days  = 7
  purge_protection_enabled    = true
  sku_name = "standard"
  tags     = local.resource_tags

}

# Get a list of all Key Management permissions
data "azurerm_key_vault_access_policy" "key_policies" {
  name = "Key Management"
}

# Grant some client all the Key Management policies offered
resource "azurerm_key_vault_access_policy" "kvap" {
  key_vault_id       = azurerm_key_vault.kv-c.id
  object_id          = var.some_service_principal_objectid
  key_permissions    = data.azurerm_key_vault_access_policy.key_policies.key_permissions
}
```

### Debug Output/Panic Output

```shell
│ Error: expected key_permissions.2 to be one of [Backup Create Decrypt Delete Encrypt Get Import List Purge Recover Restore Sign UnwrapKey Update Verify WrapKey Release Rotate GetRotationPolicy SetRotationPolicy], got update
│ 
│   with module.wqinv.module.keyvault-fs.azurerm_key_vault_access_policy.kv-devops-fs,
│   on patterns/components/keyvault/permissions.tf line 54, in resource "azurerm_key_vault_access_policy" "kvap":
│   54:   key_permissions    = data.azurerm_key_vault_access_policy.key_policies.key_permissions
│
```

### Expected Behaviour

The block `data "azurerm_key_vault_access_policy"` should have returned 'Update' instead of 'update', so that the permissions could have been applied.

### Actual Behaviour

As the capitalisation is wrong, we are unable to apply the permissions.

### Steps to Reproduce

`terraform apply`

### References

Looks like a remaining unfixed aspect of the transition to RM version 3 - see https://github.com/hashicorp/terraform-provider-azurerm/issues/16354#issuecomment-1097531707

**Comment by myc2h6o** [comment]
Hi @dvasdekis the fix is done in #17137, but unfortunately because it changes the constant values which might break other existing users, it was put behind the 4.0 flag which means it would be fixed after the 4.0 version is rolled out.
As a temporary workaround, you could iterate the list and use `title()` to fix the values:
`key_permissions = [for p in data.azurerm_key_vault_access_policy.key_policies.key_permissions : title(p)]`

**Comment by tombuildsstuff** [comment]
@myc2h6o why is that fix gated behind 4.0? That should be applied to the Data Source in a 3.x version, there's no need for that to be gated behind 4.0, and indeed we'll be auto-fixing this in the Read function of both the Key Vault Resource (and other Resources) in the not-too-distant future.

**Comment by myc2h6o** [comment]
@tombuildsstuff Got it, somehow the change was put behind 4.0, I've created #21410 to move it forward.
